Readme.md

```diff
@@ -1,22 +1,83 @@
-<!-- default badges list -->
-![](https://img.shields.io/endpoint?url=https://codecentral.devexpress.com/api/v1/VersionRange/128536219/22.1.3%2B)
-[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E4521)
-[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
-<!-- default badges end -->
-<!-- default file list -->
-*Files to look at*:
-
-* [Default.aspx](./CS/WebSite/Default.aspx) (VB: [Default.aspx](./VB/WebSite/Default.aspx))
-* [Default.aspx.cs](./CS/WebSite/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/WebSite/Default.aspx.vb))
-<!-- default file list end -->
-# ASPxGridView - How use ASPxGridLookup as FilterRow editor via FilterTemplate
+# Grid View for ASP.NET Web Forms - How to implement a filter row template and use ASPxGridLookup as an editor
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/e4521/)**
 <!-- run online end -->
 
+This example demonstrates how to create a column's filter row template, add the ASPxGridLookup control to the template, and configure the column's filter functionality.
+
+![FilterRowTemplate](filterRowTemplate.png)
+
+## Overview
+
+1. Create the [Grid View](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) control, populate it with columns, and bind it to a data source. Set the grid's [ShowFilterRow](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridViewSettings.ShowFilterRow) property to `true` to show the [filter row](https://docs.devexpress.com/AspNet/3753/components/grid-view/concepts/filter-data/filter-row).
+
+    ```aspx
+    <dx:ASPxGridView runat="server" ID="Grid" AutoGenerateColumns="False" DataSourceID="ProductsDataSource"
+        ClientInstanceName="grid" OnCustomCallback="Grid_CustomCallback" EnableViewState="false">
+        <Columns>
+            <dx:GridViewDataTextColumn FieldName="CategoryName">
+                <!-- ... -->
+            </dx:GridViewDataTextColumn>
+            <dx:GridViewDataTextColumn FieldName="ProductName" />
+            <dx:GridViewDataTextColumn FieldName="ProductSales" />
+            <dx:GridViewDataDateColumn FieldName="ShippedDate" />
+        </Columns>
+        <Settings ShowFilterRow="true" />
+    </dx:ASPxGridView>
+    ```
+
+2. Specify a column's [FilterTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewColumn.FilterTemplate) property, add the [ASPxGridLookup](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridLookup) control to the template, populate the editor with columns, and bind it to a data source.
```

Comment:
> Specify a column's [FilterTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewColumn.FilterTemplate) property, add the [ASPxGridLookup](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridLookup) **control** to the template, populate the **editor** with columns, and bind it to a data source.

This is a very good sentence, save for just one thing -- **an implied synonym**. At first, you named `ASPxGridLookup` a control, and then you refer to it as an editor. An alien reader might wonder if "control" and "editor" are the same thing in this context. A possible solution is to name `ASPxGridLookup` editor in both instances.

One more thing -- you encourage your reader to add one editor of the mentioned type to the template -- therefore, the aspect of instance is more important than the aspect of type. That's why I'd like to replace the definite article with indefinite. Add one `ASPxGridLookup` editor to the template. You could add multiple editors to this template, but for the purposes of this example, you need just one.

I hope my explanation is clear enough. Please note that I will not count this as your issue because there is room for different interpretations here.

```suggestion
2. Specify a column's [FilterTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewColumn.FilterTemplate) property, add an [ASPxGridLookup](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridLookup) editor to the template, populate the editor with columns, and bind it to a data source.
```

Make sure that you have a proper introduction for every entity you use in text.